yt/frontends/ramses/data_structures.py

```diff
@@ -175,14 +245,16 @@ def _read_particle_header(self):
 
         field_offsets = {}
         _pfields = {}
+
```

Comment:
We use the `'io'` name to denote particle types that are "generic" - i.e. if a code writes all of the different particle types to the same arrays on disk, we read those back and assume people use particle filters to create particle types they need from the `'io'` type. In some codes (e.g. Gadget), different particle types are written to different arrays on disk, so we don't have an `'io'` particle type.

We used `'io'` for Ramses because (I believe, correct me if I'm wrong) ramses star and dark matter particles are written to the same arrays on disk.

Now that you're adding sinks, we're breaking the assumption that led us to choose to have an `'io'` particle type. In principle that would lead us to drop the `'io'` particle type and try to come up with more descriptive names. However, backward compatibility concerns mean that `'io'` should continue to work.

So, for Ramses going forward, the `'io'` particle type represents all of the particles written in "standard" (i.e. what yt reads before this pull request) Ramses outputs and the `'sink'` particle type represents just sinks.

Does that make sense? Let me know if you need any further clarification.

Reply:
This makes sens. Are you OK with the way it's done in this PR?

Reply:
I haven't looked at this closely yet, I'll let you know when I do.

Reply:
Just read this over (sorry for taking so long!). This looks good to me.